pkg/controller/integration/monitor.go

```diff
@@ -81,6 +81,12 @@ func (action *monitorAction) Handle(ctx context.Context, integration *v1.Integra
 			integration.Status.IntegrationKit.Namespace, integration.Status.IntegrationKit.Name, err)
 	}
 
+	// If integration is in error and its kit is also in error then integration does not change
+	if isInIntegrationKitFailed(integration.Status) &&
```

Comment:
I am not entirely sure, but I think that if we add this, here, then, the user won't be able to edit its Integration and have a rebuild with the amended changes.

Reply:
>  Also, is there a case where the Integration get changed but it does not change the IntegrationKit ?

This is pretty much what happen every time you change a route without introducing any new component

Reply:
Can the route be the source of an IntegrationKit not building ?

Reply:
Yes. If there is no changes in the build, then, it must reuse the Integration kit. Imagine you change the definition of the Java DSL because you forgot a `;`. The IntegrationKit was good, but the runtime fails because the syntax is wrong. You would amend it with such a `;` and it would not need to rebuild a kit.

Reply:
BTW, the above scenario is a great candidate for an E2E test...